src/NUnitFramework/tests/Internal/EventListenerTextWriterTests.cs

```diff
@@ -270,64 +280,71 @@ public void TestWriteCharBufferSubstring()
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(2));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
-            Assert.That(TestResult.Outputs[1], Is.EqualTo(expected + NL));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(2));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs[1], Is.EqualTo(expected + NL));
         }
 
         [Test]
         public void TestWriteFloat()
         {
             float value = -5.0f;
 
-            TestWriter.Write(value);
-            TestWriter.WriteLine(value);
+            ListenerWriter.Write(value);
+            ListenerWriter.WriteLine(value);
 
             var expected = "-5";
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(2));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
-            Assert.That(TestResult.Outputs[1], Is.EqualTo(expected + NL));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(2));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs[1], Is.EqualTo(expected + NL));
         }
 
         [Test]
         public void TestWriteLine()
         {
-            TestWriter.WriteLine();
+            ListenerWriter.WriteLine();
 
             var expected = NL;
-            Assert.That(TestResult.Outputs.Count, Is.EqualTo(1));
-            Assert.That(TestResult.Outputs[0], Is.EqualTo(expected));
+            Assert.That(ListenerResult.Outputs.Count, Is.EqualTo(1));
+            Assert.That(ListenerResult.Outputs[0], Is.EqualTo(expected));
         }
 
         #region ITestListener implementation
 
-        private class TestListenerHelper : ITestListener
+        private class TestListenerIntercepter : ITestListener
         {
             public IList<string> Outputs { get; internal set; }
+            public ITestListener DefaultListener { get; internal set; }
 
-            public TestListenerHelper()
+            public TestListenerIntercepter(ITestListener defaultListener)
             {
+                DefaultListener = defaultListener;
                 Outputs = new List<string>();
             }
 
             void ITestListener.TestStarted(ITest test)
             {
-                // Intentionally empty
+                if (DefaultListener != null)
+                    DefaultListener.TestStarted(test);
             }
 
             void ITestListener.TestFinished(ITestResult result)
             {
-                // Intentionally empty
+                if (DefaultListener != null)
+                    DefaultListener.TestFinished(result);
             }
 
             void ITestListener.TestOutput(TestOutput output)
             {
                 Assert.IsNotNull(output);
                 Outputs.Add(output.Text);
+
+                if (DefaultListener != null)
+                    DefaultListener.TestOutput(output);
```

Comment:
Due to the original behaviour of WriteLine(char[]) sending one char at a time, sending Environment.NewLine will actually send \r and \n separately.

This causes an issue in the NUnit Visual Studio Test Adapter (see 
https://github.com/nunit/nunit3-vs-adapter/blob/749594c656b01df1495918fb07d87d3fca2ed981/src/NUnitTestAdapter/NUnitEventListener.cs#L187)

As noted in the source code, an empty string will cause SendMessage to throw an exception. Unfortunately, so does sending just \r. 

Tested under NUnit 3 Test Adapter v3.7.0.0 with VS2017.

Is this a known issue?